### ISSUE TITLE
Add check_for_api_keys_in_file to the validation chain

### ISSUE THREAD

**Issue by diox** [mozilla/addons-server#12571]
https://github.com/mozilla/addons-server/issues/12505 added a celery `chain` to orchestrate the validation tasks. #12217 and #12224 will include the scanners and yara in a `chord` inside that chain to run those in parallel.

There is one last cleanup to do: we should look moving `check_for_api_keys_in_file` task outside of `handle_upload_validation_result()`. It's not an annotation: it should run as part of the validation chain, either in the main chain, or in the `chord` that includes yara and the scanners.

**Comment by willdurand** [comment]
> There is one last cleanup to do: we should look moving `check_for_api_keys_in_file` task outside of `handle_upload_validation_result()`. It's not an annotation: it should run as part of the validation chain, either in the main chain, or in the `chord` that includes yara and the scanners.

If we add this function as a `@task` in the validation `chain` (say, after the linter and before the `chord`), we will not execute the `chord` if an api key is detected (since the function inserts a validation "error" message).

Does that sound okay?

**Comment by diox** [comment]
Yes. The developer will have to resubmit anyway, so I think this is fine.

**Comment by ioanarusiczki** [comment]
@diox Can you give some str for qa. I don't know if I got it right. I should get an error if api keys are detected (should that be inside the manifest.json) ?

**Comment by diox** [comment]
This is a refactor of https://bugzilla.mozilla.org/show_bug.cgi?id=1456797 so the same steps should apply. If a developer puts their api key anywhere in a non-binary file in their add-on, it should fail validation with an appropriate error message and revoke affected credentials automatically shortly (2 minutes is the currently configured time) after.

**Comment by AlexandraMoga** [comment]
Verified on -dev using both devhub and the signing API submission.
If the developers add their API credentials in any of the add-on files, the following validation error will be raised:
![image](https://user-images.githubusercontent.com/31961530/68119498-8dc91700-ff0b-11e9-8b3e-947ae7372a09.png)

Subsequently, the respective API keys will be revoked and the user can generate new ones.

@diox for the API submissions, the validation error is not returned in the response body, so there is no immediate way of telling what caused the failure. There is, however, a link to the validation results which can be opened in devhub, where the reason for the submission failure is clear.  Would it be possible to show this error message directly in the response?

**Comment by diox** [comment]
Changing how validation errors are returned in the API is a big task, out of scope here.
